Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabSaveLoadHandler.cpp

```diff
@@ -129,13 +143,255 @@ namespace AzToolsFramework
             settings->m_saveLocation = path;
         }
 
-        void PrefabSaveHandler::HandleSourceFileType(AZStd::string_view sourceFilePath, AZ::EntityId parentId, AZ::Vector3 position) const
+        // Drag and drop overrides:
+        int PrefabSaveHandler::GetDragAndDropEventsPriority() const
+        {
+            return 10; // higher than 0 so this class gets ahead of other built in 
+        }
+
+        // viewport dragging:
+        void PrefabSaveHandler::DragEnter(QDragEnterEvent* event, AzQtComponents::DragAndDropContextBase& context)
         {
-            auto instantiatePrefabOutcome = s_prefabPublicInterface->InstantiatePrefab(sourceFilePath, parentId, position);
+            if (event->isAccepted())
+            {
+                return;
+            }
 
-            if (!instantiatePrefabOutcome.IsSuccess())
+            if (CanDragAndDropData(event->mimeData()))
             {
-                WarningDialog("Prefab Instantiation Error", instantiatePrefabOutcome.GetError());
+                event->accept();
+                event->setDropAction(Qt::DropAction::CopyAction);
+            }
+
+            (void)context;
+
+        }
+
+        void PrefabSaveHandler::DragMove(QDragMoveEvent* event, AzQtComponents::DragAndDropContextBase& context)
+        {
+            if (event->isAccepted())
+            {
+                return;
+            }
+
+            if (CanDragAndDropData(event->mimeData()))
+            {
+                event->accept();
+                event->setDropAction(Qt::DropAction::CopyAction);
+            }
+            (void)context;
+        }
+
+        void PrefabSaveHandler::DragLeave(QDragLeaveEvent* event)
+        {
+            if (event->isAccepted())
+            {
+                return;
+            }
+        }
+
+        void PrefabSaveHandler::Drop(QDropEvent* event, AzQtComponents::DragAndDropContextBase& context)
+        {
+            using namespace AzToolsFramework;
+            using namespace AzToolsFramework::AssetBrowser;
+            using namespace AzQtComponents;
+
+            if (event->isAccepted())
+            {
+                return;
+            }
+
+            if (ViewportDragContext* viewportContext = azrtti_cast<ViewportDragContext*>(&context);viewportContext)
```

Comment:
Does this line need the `;viewportContext` ? Doesn't it check for nullptr by having the assignment inside the condition?

Reply:
So, technically it works, because it will evaluate to a value that can be converted to a boolean.  I'll remove this and the below if nobody is upset that we are coercing the return from an if to a boolean.  We do this all the time with nullptr all over the code, so I don't see a huge issue.

Reply:
Sounds good that makes sense. Thanks for the clarification!